src/main/io/gimbal_control.c

```diff
@@ -0,0 +1,329 @@
+    gimbalCmdOut.pitch = constrain(pitch, GIMBAL_PITCH_MIN * gimbalTrackConfig()->gimbal_pitch_low_limit / 100,
+                                          GIMBAL_PITCH_MAX * gimbalTrackConfig()->gimbal_pitch_high_limit / 100);
+    gimbalCmdOut.yaw   = constrain(yaw, GIMBAL_YAW_MIN * gimbalTrackConfig()->gimbal_yaw_limit / 100,
+                                        GIMBAL_YAW_MAX * gimbalTrackConfig()->gimbal_yaw_limit / 100);
+
+    gimbalCmdOut.mode = gimbalTrackConfig()->gimbal_stabilisation;
+    gimbalCmdOut.sens = gimbalTrackConfig()->gimbal_sensitivity;
+
+    uint16_t crc = gimbalCrc((uint8_t *)&gimbalCmdOut, sizeof(gimbalCmdOut) - 2);
+
+    gimbalCmdOut.crc = crc;
+
+    return true;
+}
+
+// Gimbal updates should be sent at 100Hz or the gimbal will self center after approx. 2 seconds
+void gimbalUpdate(timeUs_t currentTimeUs)
+{
+    static enum {GIMBAL_OP1, GIMBAL_OP2, GIMBAL_CMD} gimbalParseState = GIMBAL_OP1;
+    static timeUs_t lastRxTimeUs = 0;
+
+    if (!gimbalSerialPort) {
+        setTaskEnabled(TASK_GIMBAL, false);
+        return;
+    }
+
+    // Read bytes from the VTX gimbal serial data stream
+
+    uint32_t bytes =  serialRxBytesWaiting(gimbalSerialPort);
+
+    if (bytes > 0) {
+        lastRxTimeUs = currentTimeUs;
+        while (bytes--) {
+            uint8_t inData = serialRead(gimbalSerialPort);
+
+            // If the packet is a gimbalCmd_t structure then parse, otherwise pass through
+            switch (gimbalParseState) {
+            default:
+            case GIMBAL_OP1:
+                if (inData == (GIMBAL_CMD_S & 0xff)) {
+                    gimbalParseState = GIMBAL_OP2;
+                } else {
+                    serialWrite(gimbalSerialPort, inData);
+                }
+                break;
+            case GIMBAL_OP2:
+                if (inData == ((GIMBAL_CMD_S >> 8) & 0xff)) {
+                    gimbalParseState = GIMBAL_CMD;
+                    gimbalInCount = sizeof(gimbalCmdIn.u.gimbalCmd.opcode);
+                } else {
+                    serialWrite(gimbalSerialPort, GIMBAL_CMD_S && 0xff);
+                    serialWrite(gimbalSerialPort, inData);
+                    gimbalParseState = GIMBAL_OP1;
+                }
+                break;
+            case GIMBAL_CMD:
+                gimbalCmdIn.u.bytes[gimbalInCount++] = inData;
+                if (gimbalInCount == sizeof(gimbalCmdIn.u.gimbalCmd)) {
+                    uint16_t crc = gimbalCrc((uint8_t *)&gimbalCmdIn, sizeof(gimbalCmdIn) - 2);
+                    // Only use the data if the CRC is correct
+                    if (gimbalCmdIn.u.crc == crc) {
+                        gimbalCmdOut = gimbalCmdIn.u.gimbalCmd;
+                        gimbalSet(gimbalCmdIn.u.gimbalCmd.roll, gimbalCmdIn.u.gimbalCmd.pitch, gimbalCmdIn.u.gimbalCmd.yaw);
+                        serialWriteBuf(gimbalSerialPort, (uint8_t *)&gimbalCmdOut, sizeof(gimbalCmdOut));
+                    }
```

Comment:
Really not sure here, but maybe packet with invalid crc should be forwarded - modifying correctly recognized packets, forwarding everything else.